app/models/dossier.rb

```diff
@@ -645,56 +645,79 @@ def build_attestation
     end
   end
 
-  def expired_keep_track!
-    if keep_track_on_deletion?
-      DeletedDossier.create_from_dossier(self, :expired)
-      log_automatic_dossier_operation(:supprimer, self)
+  def expired_keep_track_and_destroy!
+    begin
+      transaction do
+        if keep_track_on_deletion?
+          DeletedDossier.create_from_dossier(self, :expired)
+          log_automatic_dossier_operation(:supprimer, self)
+        end
+        destroy!
+      end
+      id
+    rescue
+      nil
```

Comment:
Pinaillage de style : il me semblait qu'on pouvait mettre le `rescue` dans le corps de la méthode ; genre :

```suggestion
  def expired_keep_track_and_destroy!
    transaction do
      if keep_track_on_deletion?
        DeletedDossier.create_from_dossier(self, :expired)
        log_automatic_dossier_operation(:supprimer, self)
      end
      destroy!
    end
    id
  rescue
    nil
```